python/cuspatial/cuspatial/core/spatial/distance.py

```diff
@@ -101,14 +101,41 @@ def haversine_distance(p1: GeoSeries, p2: GeoSeries):
     Parameters
     ----------
     p1: GeoSeries
-        Series of points
+        Series of points as floats
     p2: GeoSeries
-        Series of points
+        Series of points as floats
 
     Returns
     -------
     result : cudf.Series
         The distance between pairs of points between `p1` and `p2`
+    >>> import cudf
+    >>> import cuspatial
+    >>> a = {"latitude":[0.0,0.0,1.0,1.0],
+    ...      "longitude": [0.0,1.0,0.0,1.0]}
+    >>> df = cudf.DataFrame(data=a)
+
+    >>> # Create cuSpatial GeoSeries from cuDF Dataframe
+    >>> cuGeoSeries = cuspatial.GeoSeries.from_points_xy(
+    ...     df[['longitude', 'latitude']].interleave_columns()
+    ... )
+
+    >>> # Create Comparator cuSpatial GeoSeries from a comparator point
+    >>> df['compare_lat'] = 2.0
+    >>> df['compare_lng'] = 2.0
```

Comment:
As a DF novice, it wasn't obvious to me that this broadcasts (2.0, 2.0) to all 4 rows of the existing DF. I was trying to figure out why there wasn't an error from trying to compute distances between a length 4 and a length 1 series of points.

Reply:
You're absolutely right!  Great catch.  I added a comment to the example to bridge this gap.